tests/visualization_tests/matplotlib_tests/test_contour.py

```diff
@@ -150,3 +154,70 @@ def test_plot_contour_customized_target_name(params: List[str]) -> None:
         assert figure.shape == (len(params), len(params))
         for i in range(len(params)):
             assert figure[i][0].yaxis.label.get_text() == list(params)[i]
+
+
+def test_plot_contour_log_scale_and_str_category() -> None:
+
+    # If the search space has two parameters, plot_contour generates a single plot.
+    study = create_study()
+    study.add_trial(
+        create_trial(
+            value=0.0,
+            params={"param_a": 1e-6, "param_b": "100"},
+            distributions={
+                "param_a": FloatDistribution(1e-7, 1e-2, log=True),
+                "param_b": CategoricalDistribution(["100", "101"]),
+            },
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=1.0,
+            params={"param_a": 1e-5, "param_b": "101"},
+            distributions={
+                "param_a": FloatDistribution(1e-7, 1e-2, log=True),
+                "param_b": CategoricalDistribution(["100", "101"]),
+            },
+        )
+    )
+
+    figure = plot_contour(study)
+
+    # Take 5% axis padding into account.
+    np.testing.assert_allclose(figure.get_xlim(), [1e-6, 1e-5], atol=5e-2)
+    np.testing.assert_allclose(figure.get_ylim(), [0.0, 1.0], atol=5e-2)
```

Comment:
It's not a strong opinion but I think a test case for three parameters well covers  because this test doesn't focus on a number of parameters but focus on padding for categorical/logarithmic scale.

Reply:
You're right, this would make an unnecessary duplication. Removed in af25f5931593285548c4e263d6cd8b1d776ac7a2.